src/utils/hall.js

```diff
@@ -0,0 +1,43 @@
+export const allSeats = (arr1, arr2) => {
+  const available = arr1.map((seat) => {
+    return {
+      seat,
+      available: true,
+      row: seat.slice(0, 1),
+      seatNumber: seat.slice(1),
+    };
+  });
+  const taken = arr2.map((seat) => {
+    return {
+      seat,
+      available: false,
+      row: seat.slice(0, 1),
+      seatNumber: seat.slice(1),
+    };
+  });
+  const seats = [...available, ...taken];
+  seats.sort((a, b) => a.seat.localeCompare(b.seat));
+  return seats;
+};
+
+export const numberOfSeatsInRow = (array) => {
+  const listOfSeats = array.map((el) => el.seatNumber);
+  const uniqueSeats = listOfSeats.filter(
+    (seat, i, arr) => arr.indexOf(seat) === i
+  );
+  return uniqueSeats.length;
+};
+
+export const hallPlan = (array) => {
```

Comment:
Likewise for these function, argument names like `array` are too generic! We should give more hints to what's going on here 😄